lib/models/notification.ts

```diff
@@ -49,55 +49,48 @@ export const NotificationBase = t.interface({
 
 // Email Notification
 
-export const NotificationChannelEmail = t.interface({
-  [NotificationChannelEnum.EMAIL]: t.intersection([
-    t.interface({
-      addressSource: NotificationAddressSource,
-      toAddress: EmailAddress
-    }),
-    t.partial({
-      fromAddress: EmailAddress
-    })
-  ])
-});
+export const NotificationChannelEmail = t.intersection([
+  t.interface({
+    addressSource: NotificationAddressSource,
+    toAddress: EmailAddress
+  }),
+  t.partial({
+    fromAddress: EmailAddress
+  })
+]);
 export type NotificationChannelEmail = t.TypeOf<
   typeof NotificationChannelEmail
 >;
 
 export const EmailNotification = t.interface({
   ...NotificationBase.props,
-  channel: NotificationChannelEmail
+  channel: NotificationChannelEmail,
+  type: t.literal(NotificationChannelEnum.EMAIL)
 });
 export type EmailNotification = t.TypeOf<typeof EmailNotification>;
 
 // Webhook Notification
 
 export const NotificationChannelWebhook = t.interface({
-  [NotificationChannelEnum.WEBHOOK]: t.interface({
-    url: HttpsUrl
-  })
+  url: HttpsUrl
 });
 export type NotificationChannelWebhook = t.TypeOf<
   typeof NotificationChannelWebhook
 >;
 
 export const WebhookNotification = t.interface({
   ...NotificationBase.props,
-  channel: NotificationChannelWebhook
+  channel: NotificationChannelWebhook,
```

Comment:
same here, it looks like having the channel specific attributes inside `channel` can be avoided, getting rid of the extra type `NotificationChannelWebhook`

Reply:
two reasons. 1. there are some functions (like [this one](https://github.com/teamdigitale/digital-citizenship-functions/blob/97279ea7dd5ce53396ad30afb179de56e72d9c1f/lib/created_message_queue_handler.ts#L161)) that works on the channel metadata. we would need to define a type for these functions which has the same shape of NotificationChannelWebhook 2. [toBaseType](https://github.com/teamdigitale/digital-citizenship-functions/pull/182/files#diff-f77a48800133699f87c40a0d01fef39eR138) is supposed to return a "complete" notification (so it extracts the channel metadata aside NotificationBase properties). if we flatten the metadata we will complicate the logic to extract a Notification from a RetrievedNotification.